src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2824,7 +2824,7 @@ private ParseTree parseAssignment(Expression expressionIn) {
 
     if (isStartOfAsyncArrowFunction(left)) {
       // re-evaluate as an async arrow function.
-      resetScanner(left);
+      resetScanner(start);
```

Comment:
I'll take another look, but `left` was no longer referring to `async` - it now referred to the beginning of the call `(`. That's why this was changed.